core/src/main/scala/akka/kafka/internal/CommittingProducerSinkStage.scala

```diff
@@ -164,7 +168,18 @@ private final class CommittingProducerSinkStageLogic[K, V, IN <: Envelope[K, V,
     case CommittingProducerSinkStage.CommitNow => commit(Interval)
   }
 
-  private def collectOffset(count: Int, offset: Committable): Unit = {
+  private def collectOffset(count: Int, offset: Committable): Unit =
+    (stage.committerSettings.trigger, deferredOffset) match {
+      case (FirstObserved, _) =>
+        addToBatch(count, offset)
+      case (DeferToNextCommit, None) =>
+        deferredOffset = Some(DeferredOffset(offset, count))
+      case (DeferToNextCommit, Some(DeferredOffset(dOffset, dCount))) if dOffset != offset =>
+        deferredOffset = Some(DeferredOffset(offset, count))
+        addToBatch(dCount, dOffset)
+    }
+
+  private def addToBatch(count: Int, offset: Committable): Unit = {
     awaitingProduceResult -= count
```

Comment:
The `count` is not about the batch, you may update its value independently. That makes the `DeferredOffset` type go away.

Reply:
I see.